### ISSUE TITLE
Elastic Index ignores documents created using Cassandra UPDATE

### ISSUE THREAD

**Issue by shohamlevy** [strapdata/elassandra#50]
I am using elassandra-2.1.1-17

When creating new documents using INSERT, Elastic indexes them. But when  using UPDATE to create the new documents - Elastic ignores them. See example:

``` CQL
cqlsh > CREATE KEYSPACE IF NOT EXISTS test
    WITH replication = {
                           'class': 'NetworkTopologyStrategy',
                           'dc1': '1'
                       }
    AND durable_writes = true;

cqlsh > CREATE TABLE test.t1 (
    name text,
    id int,
    nicks set<text>,
    PRIMARY KEY (name, id)
);
```

``` bash
$> curl -XDELETE "localhost:9200/test_index?pretty=true" 

$> curl -XPUT "localhost:9200/test_index?pretty=true" -d \
'{
    "settings": {
        "keyspace":"test"
    },
    "mappings": {
            "t1": {
                "properties": {
                    "name": {"type": "string", "cql_collection": "singleton", "index": "not_analyzed", "cql_primary_key_order": 0, "cql_partition_key": true},
                    "id": {"type": "integer", "cql_collection": "singleton", "index": "no", "cql_primary_key_order": 1},
                    "nicks": {"type": "string", "cql_collection": "set", "index": "not_analyzed"}
                }
            }
    }
}' 
```

Now add a couple of documents:

``` CQL
cqlsh> UPDATE test.t1 SET nicks = nicks + {'abc'} WHERE name='Moses' AND id=14;
cqlsh> INSERT INTO test.t1 (name, id, nicks) VALUES ('Jerry', 33, {'jj', 'jlo'});
cqlsh> SELECT * FROM test.t1 ;
 name  | id | nicks
-------+----+---------------
 Moses | 14 |       {'abc'}
 Jerry | 33 | {'jj', 'jlo'}
(2 rows)
```

Check results for Elastic (you only see 1 entry, the one added using INSERT):

``` bash
$> curl -XGET "localhost:9200/test_index/t1/_search?pretty=true"
{
  "took" : 16,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  },
  "hits" : {
    "total" : 1,
    "max_score" : 1.0,
    "hits" : [ {
      "_index" : "test_index",
      "_type" : "t1",
      "_id" : "[\"Jerry\",33]",
      "_score" : 1.0,
      "_source":{"name":"Jerry","id":33,"nicks":["jj","jlo"]}
    } ]
  }
}
```

**Comment by shohamlevy** [comment]
I run some more checks. Apparently ElasticSearch (for Elssandra) does not like to work with Cassandra SET, but only with LIST.

Changing 'nicks' from SET of Strings to LIST of Strings solved some ES query issues I had. **But it still does not index UPDATE entries.**

**Comment by vroyer** [comment]
Hi,
Yes, there is a bug when updating a collection with a clustering key.
This is fixed for the next release delivered by the end of this week.
Thanks for reporting this issue.
Vincent.

**Comment by vroyer** [comment]
This is fixed in the last release 2.1.1-18.

**Comment by shohamlevy** [comment]
Checked, and now it works. Thanks!
